pkgs/applications/science/chemistry/marvin/default.nix

```diff
@@ -0,0 +1,40 @@
+{ stdenv, lib, fetchurl, dpkg, jre, makeWrapper }:
+
+let
+  pname = "marvin";
+  version = "19.1.0";
+  src =
+    fetchurl {
+      name = "marvin-${version}.deb";
+      url = "http://dl.chemaxon.com/marvin/${version}/marvin_linux_19.1.deb";
+      sha256 = "1ccsimfvms5q4prjyk6sg5hsc3hkcjjfq3gl7jjm8dgd2173zzyc";
+    };
+in stdenv.mkDerivation {
+  inherit version;
+  inherit pname;
+  inherit src;
+  buildInputs = [ dpkg makeWrapper ];
+  unpackPhase = ''
+    mkdir pkg/opt -p
+    dpkg-deb -x $src pkg/opt
+  '';
+  installPhase = ''
+    cp -r pkg/* $out
+    mkdir $out/share/applications -p
+    mkdir $out/share/pixmaps
+    mkdir $out/bin
+    ln -sf $out/opt/chemaxon/marvinsuite/bin/msketch $out/bin/msketch
+    ln -sf $out/opt/chemaxon/marvinsuite/bin/mview $out/bin/mview
```

Comment:
This does not work in an empty PATH. 
```
 PATH= $buildInputs/bin/mview 
/nix/store/jb299sb3abjqfh4zm7qpkd6ir0jl8jfc-marvin-19.1.0/bin/mview: line 362: basename: No such file or directory
/nix/store/jb299sb3abjqfh4zm7qpkd6ir0jl8jfc-marvin-19.1.0/bin/mview: line 363: dirname: No such file or directory
/nix/store/jb299sb3abjqfh4zm7qpkd6ir0jl8jfc-marvin-19.1.0/bin/mview: line 378: dirname: No such file or directory
/nix/store/jb299sb3abjqfh4zm7qpkd6ir0jl8jfc-marvin-19.1.0/bin/mview: line 379: basename: No such file or directory
No suitable Java Virtual Machine could be found on your system.
The version of the JVM must be at least 1.8.
Please define INSTALL4J_JAVA_HOME to point to a suitable JVM.
```
It would be preferable to wrap these executables with the jre (and coreutils apparently). random example:
https://github.com/NixOS/nixpkgs/blob/b07e1563fe4cafd5e375a5861e42c10e0406b792/pkgs/applications/audio/abcde/default.nix#L38-L43
Or if running a jar is enough:
https://github.com/NixOS/nixpkgs/blob/b07e1563fe4cafd5e375a5861e42c10e0406b792/pkgs/applications/graphics/astah-community/default.nix#L52-L53

Reply:
The executing command, when the variables are expanded, is incredibly complex, e.g. here is what it is when I've extracted it to a local directory at /home/fusion809/Programs/Deb/marvin: https://gist.github.com/fusion809/188669fd39c100a2cfadc42c72774fb4.

Reply:
Oh thank goodness, I figured it out! Just had to find the docs for makeWrapper.

Reply:
It still does not work for me in an empty PATH.
```
[nix-shell:~/.cache/nix-review/pr-53839]$ PATH= $buildInputs/bin/mview
/nix/store/vs9h3hws3slbrliryxi80abmf48dws4j-marvin-19.1.0/opt/chemaxon/marvinsuite/bin/mview: line 362: basename: No such file or directory
/nix/store/vs9h3hws3slbrliryxi80abmf48dws4j-marvin-19.1.0/opt/chemaxon/marvinsuite/bin/mview: line 363: dirname: No such file or directory
/nix/store/vs9h3hws3slbrliryxi80abmf48dws4j-marvin-19.1.0/opt/chemaxon/marvinsuite/bin/mview: line 378: dirname: No such file or directory
/nix/store/vs9h3hws3slbrliryxi80abmf48dws4j-marvin-19.1.0/opt/chemaxon/marvinsuite/bin/mview: line 379: basename: No such file or directory
/nix/store/vs9h3hws3slbrliryxi80abmf48dws4j-marvin-19.1.0/opt/chemaxon/marvinsuite/bin/mview: line 66: expr: No such file or directory
/nix/store/vs9h3hws3slbrliryxi80abmf48dws4j-marvin-19.1.0/opt/chemaxon/marvinsuite/bin/mview: line 67: expr: No such file or directory
/nix/store/vs9h3hws3slbrliryxi80abmf48dws4j-marvin-19.1.0/opt/chemaxon/marvinsuite/bin/mview: line 81: awk: No such file or directory
/nix/store/vs9h3hws3slbrliryxi80abmf48dws4j-marvin-19.1.0/opt/chemaxon/marvinsuite/bin/mview: line 82: rm: No such file or directory
/nix/store/vs9h3hws3slbrliryxi80abmf48dws4j-marvin-19.1.0/opt/chemaxon/marvinsuite/bin/mview: line 83: mv: No such file or directory
/nix/store/vs9h3hws3slbrliryxi80abmf48dws4j-marvin-19.1.0/opt/chemaxon/marvinsuite/bin/mview: line 85: sed: No such file or directory
/nix/store/vs9h3hws3slbrliryxi80abmf48dws4j-marvin-19.1.0/opt/chemaxon/marvinsuite/bin/mview: line 88: chmod: No such file or directory
/nix/store/vs9h3hws3slbrliryxi80abmf48dws4j-marvin-19.1.0/opt/chemaxon/marvinsuite/bin/mview: line 66: expr: No such file or directory
/nix/store/vs9h3hws3slbrliryxi80abmf48dws4j-marvin-19.1.0/opt/chemaxon/marvinsuite/bin/mview: line 67: expr: No such file or directory
/nix/store/vs9h3hws3slbrliryxi80abmf48dws4j-marvin-19.1.0/opt/chemaxon/marvinsuite/bin/mview: line 81: awk: No such file or directory
/nix/store/vs9h3hws3slbrliryxi80abmf48dws4j-marvin-19.1.0/opt/chemaxon/marvinsuite/bin/mview: line 82: rm: No such file or directory
/nix/store/vs9h3hws3slbrliryxi80abmf48dws4j-marvin-19.1.0/opt/chemaxon/marvinsuite/bin/mview: line 83: mv: No such file or directory
/nix/store/vs9h3hws3slbrliryxi80abmf48dws4j-marvin-19.1.0/opt/chemaxon/marvinsuite/bin/mview: line 85: sed: No such file or directory
/nix/store/vs9h3hws3slbrliryxi80abmf48dws4j-marvin-19.1.0/opt/chemaxon/marvinsuite/bin/mview: line 88: chmod: No such file or directory
No suitable Java Virtual Machine could be found on your system.
The version of the JVM must be at least 1.8.
Please define INSTALL4J_JAVA_HOME to point to a suitable JVM.
```
You probably need to add something like `--prefix PATH : ${makeBinPath [ coreutils awk sed ]}`.
And the INSTALL4J_JAVA_HOME thing does not seem to work.
By the way, I find strange that a startup wrapper needs rm...

Reply:
I added that to the wrapper commands and now building it gives:

```bash
rror: undefined variable 'makeBinPath' at /data/GitHub/mine/packaging/nixpkgs.marvin-pr/pkgs/applications/science/chemistry/marvin/default.nix:21:117
```

Reply:
makeBinPath is in lib

Reply:
Replacing `makeBinPath` with `stdenv.lib.makeBinPath` makes the build fail, with the error:

```bash
error: undefined variable 'coreutils' at /data/GitHub/mine/packaging/nixpkgs.marvin-pr/pkgs/applications/science/chemistry/marvin/default.nix:21:142
```

Sorry, I'm sure these are real stupid concerns, but I'm a beginner and I do not know why this is happening.

Reply:
coreutils needs to be added to the arguments in the first line.

Reply:
Thank you, I do believe I have fixed it. I tested with an empty PATH and it launched.

Reply:
nice !